src/gui/clips/SampleClipView.cpp

```diff
@@ -183,16 +182,22 @@ void SampleClipView::mouseDoubleClickEvent( QMouseEvent * )
 {
 	QString af = SampleLoader::openAudioFile();
 
-	if ( af.isEmpty() ) {} //Don't do anything if no file is loaded
-	else if (af == m_clip->m_sample.sampleFile())
-	{	//Instead of reloading the existing file, just reset the size
-		int length = static_cast<int>(m_clip->m_sample.sampleSize() / Engine::framesPerTick());
-		m_clip->changeLength(length);
+	// Don't do anything if no file is loaded
+	if (af.isEmpty()) { return; }
+	
+	if (af == m_clip->m_sample.sampleFile())
+	{
+		// Don't reload the same file again. Just reset the size.
+		m_clip->changeLengthToSampleLength();
 	}
 	else
```

Comment:
You could combine this like so:
```suggestion
	else if (auto sampleBuffer = SampleLoader::createBufferFromFile(af); sampleBuffer != SampleBuffer::emptyBuffer()) 
```

Reply:
Thanks for the suggestion but I find it more readable and quicker to comprehend as is.